src/librustc/ty/constness.rs

```diff
@@ -68,14 +69,54 @@ impl<'tcx> TyCtxt<'tcx, 'tcx> {
 
 
 pub fn provide<'tcx>(providers: &mut Providers<'tcx>) {
-    /// only checks whether the function has a `const` modifier
+    fn is_const_evaluatable(tcx: TyCtxt<'tcx, 'tcx>, def_id: DefId) -> bool {
+        // Intrinsics promotion whitelist is here to check const evaluability at the
```

Comment:
Is this a *promotion* whitelist or a "const evaluatable" whitelist? The name says one thing and the comment another thing.

Reply:
Ah, seems like you forgot to update the comment when you renamed the function.

@oli-obk do we have a test like this to ensure that intrinsics are never promoted?
```rust
#![feature(core_intrinsics)]
fn main() {
    let x: &'static usize = &std::intrinsics::size_of::<i32>(); //~ ERROR temporary value dropped while borrowed
}
```

Reply:
I didn't add that test to remote, it is in my local right now. Before that, this PR has one stage 1 failure (you can check the build log added by highfive)

Reply:
@RalfJung 	yes we have such tests, but the only one that could ever be user-facing is https://github.com/rust-lang/rust/blob/c21fbfe7e310b9055ed6b7c46b7d37b831a516e3/src/test/ui/consts/const-eval/transmute-const-promotion.rs#L6 since all other intrinsics have stable wrappers

Reply:
Sure, just wanted to make sure the other intrinsics (in particular the safe ones) are also covered, just in case.

I can't find any such test?

Reply:
we don't test intrinsics directly, just via their wrappers... so I guess more tests would be good?

Reply:
@vertexclique could you add a test like what I suggested [above](https://github.com/rust-lang/rust/pull/61835#discussion_r294073426)? Not sure if that should be a new test or added to the only existing intrinsic-promotion test `transmute-const-promotion` (which should be renamed in that case).